conan/locks/backend_fasteners.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import os
+import threading
+from contextlib import contextmanager
+from io import StringIO
+
+import fasteners
+
+from conan.locks.backend import LockBackend
+from conan.locks.exceptions import AlreadyLockedException
+
+log = logging.getLogger(__name__)
+
+
+class RWLock(object):
+    def __init__(self, resource: str, interprocess_lock: str):
+        self.w_lock = threading.Lock()
```

Comment:
I am checking possible multi-threading race conditions, and given the structure of the program, typical operation (parallel uploads, parallel downloads), etc., we can guarantee by the business logic that multithreaded operations within the same process will not collide or produce race conditions with itself. Thus, lets simplify this, leaving only the interprocess lock, the simplicity will pay off